### ISSUE TITLE
Crash when canceling resolution change

### ISSUE THREAD

**Issue by froyton** [larsiusprime/tdrpg-bugs#551]
Operating System: Windows 7 SP1 (x86)
Version #:  DQ1 0.2.6 (public HD demo)
Extra info:  running in compatibility mode (XP SP3) because video was not working initially, but running the game in compatibility mode makes video work for me

Steps to reproduce:
1. From main menu, click Options
2. Click Video
3. Select a different resolution (any one should do)
4. Click Cancel
5. Message box:  "Error! Scale must be positive & > 0! (Scale was = 0)

**Comment by larsiusprime** [comment]
Interesting. What happens if you go to your save folder (can find it by plugging in %APPDATA%Roaming\LevelUpLabs\DefendersQuest the explorer bar and hitting enter), opening the "options.xml" file, and edit the window size manually?

look for:
```
<video>
		<screen>
			<border>
				<width value="0"/>
				<height value="0"/>
			</border>
			<offset>
				<x value="0"/>
				<y value="0"/>
			</offset>
			<size value="800,600"/>
		</screen>
```

The "size" tag below the "screen" tag is the one you want. (there's another one lower down, inside the "mode" tag that is NOT what you want).

Enter a size like "1024,768" and save the file, then restart the game. What happens?

(If it crashes or errors and you can no longer start it up safely, delete the options.xml and that will reset to default settings).

**Comment by froyton** [comment]
Well, just to clarify, within the game I can set a new resolution, save the setting and resume without a problem.

The crash occurs if I go to the options menu, select a new resolution (which changes the window size right then and there) and if I click "cancel" without saving the setting.

I would check your latest demo to see if the crash is still present, but unfortunately the binary is crashing as soon as I run it.  :(  Unless other people are reporting crashing on startup, I wouldn't put much stock in that.  I might actually have some issues on my system because I ran into a similar problem with another game recently.  I'll download the demo onto another more reliable computer and reply back here soon.

**Comment by froyton** [comment]
Ok, I was able to get the demo running on my work laptop (shh don't tell) so I can continue testing.  Windows 8.1 x86_64

I am able to replicate the originally reported issue when using the latest build (0.3.6 as of this writing).  I probably didn't explain it very well so I'll try to elaborate.  What happens is, I'm in the video options, and I have a list of resolutions I can choose from.  If I click on any one of those, it will immediately change the window size.  Then I can either click "accept" or "cancel".

If I click "accept", I continue on my merry way with the new resolution and it works fine.

If I click "cancel", I would expect it to revert to the old resolution, but instead it crashes and throws that error at me.

I hope that explains it better!

**Comment by larsiusprime** [comment]
Okay, interesting. I'll try to reproduce.

**Comment by larsiusprime** [comment]
Wow! That was really easy to reproduce. Can't believe I haven't fixed this yet.

**Comment by larsiusprime** [comment]
Okey dokes, this will be fixed in version 0.3.8
